### PR TITLE
Fix ScanPy default stdout

### PR DESCRIPTION
**Pull request by mtekman**
FOR CONTRIBUTOR:
* [ ] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [X] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

The motivation here is to reduce the training time of the *PBMC 3K with ScanPy* training material, where there is a large time bottleneck in performing inspection on the generated H5AD sets. 

Instead of asking users to query their H5AD sets for general information, this PR aims to just print this nicely within the stdout window within the H5AD dataset in the history. The stdout at the moment contains debugging information, so this will be redirected to a seperate hidden output (so it can be still tested against, but not visible to the user).

### CODE REVIEW

File: tools/scanpy/macros.xml [hunk 1]
```diff
@@ -21,9 +21,10 @@
     </xml>
     <token name="@CMD@"><![CDATA[
 cp '$adata' 'anndata.h5ad' &&
-cat '$script_file' &&
-python '$script_file' &&
-ls .
+cat '$script_file' > '$hidden_output' &&
+python '$script_file' >> '$hidden_output' 2>&1 &&
```
Comment (mvdbeek) [inline tools/scanpy/macros.xml]: Redirecting the stderr as you do here will lead to a tool will be impossible to debug.Do you really need to redirect stderr as well ?

Comment (mtekman) [reply tools/scanpy/macros.xml]: sadly yes, but I will try to find a way to do that within python

Comment (mtekman) [reply tools/scanpy/macros.xml]: @mvdbeek A bit cleaner now https://github.com/galaxyproject/tools-iuc/pull/2868/commits/2ad7f8783b1ecfc35998260d2848b696a025af0d

**Comment by bgruening** [comment]
@mtekman it is failing in Docker. If you run your tests with `planemo t --docker` they will fail. Without Docker it works.

**Comment by bgruening** [comment]
How much time do you spend here? Is the debug output that bad? Can you not add the information that you need ontop of the current output?

**Comment by mtekman** [comment]
Ah I forgot docker, thanks I will try with that. The debug output throws future depreciation function output taking up 6 lines in the stdout window. That stdout window could be better put to use giving actual h5ad metrics instead. It would also align it with the RaceId tos which give info about the binary RDS object.

**Comment by mtekman** [comment]
Found the problem. Fixing now

**Comment by mtekman** [comment]
@mtekman Future Reference:

* Test against the docker image if you can't reproduce Travis bugs
* Don't redirect stderr somewhere, otherwise its impossible to debug (as mentioned)
* Isolate just one test case and incrementally simplify the CMD until bug is found

In this case the problem was that Python in the Docker image had trouble writing out to `utf-8`, and was set to `ascii` by default. I'm guessing on my system this is different.

File: tools/scanpy/macros.xml [hunk 2]
```diff
@@ -41,11 +46,21 @@ adata = sc.read('anndata.h5ad')
     </token>
     <xml name="anndata_outputs">
         <data name="anndata_out" format="h5ad" from_work_dir="anndata.h5ad" label="${tool.name} (${method.method}) on ${on_string}: Annotated data matrix"/>
+        <data name="hidden_output" format="txt" label="Log file" hidden="true" >
+            <!-- The purpose of this is to consolidate all unnecessary logging information into a hidden
+                 output file that:
```
Comment (bgruening) [inline tools/scanpy/macros.xml]: Why do we not make this output a normal output but optional as we do usually.

Comment (mtekman) [reply tools/scanpy/macros.xml]: because the debug info is only useful to the developer and the user will never want to see it IMO

Comment (mtekman) [reply tools/scanpy/macros.xml]: but it will mess with the history numbering....
Hmm.... okay give me an hour or so to patch it your way ;-)

**Comment by bgruening** [comment]
Sorry, for the delay ....

**Pull request merged**